src/conda_project/_conda_lock.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (C) 2022-2024 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+
+from pathlib import Path
+from typing import AbstractSet, Dict, List, Optional, Sequence
+
+import conda_lock
+from conda_lock.conda_lock import (
+    MetadataOption,
+    PathLike,
+    TKindAll,
+    default_virtual_package_repodata,
+)
+from conda_lock.conda_lock import make_lock_files as _make_lock_files
+from conda_lock.conda_lock import make_lock_spec as _make_lock_spec
+from conda_lock.models.lock_spec import LockSpecification
+from packaging.version import parse
+
+
+def is_conda_lock_3() -> bool:
+    return parse(conda_lock.__version__) >= parse("3.0.0")
+
+
+if is_conda_lock_3():  # pragma: no cover
+    from conda_lock.lookup import DEFAULT_MAPPING_URL  # pragma: no cover
```

Comment:
Small nit. Having this conditional import here detaches it from its usage in two places below. One alternative would be to replace this with a wrapper function called e.g. `get_mapping_url`, which encapsulates this import and returns either `DEFAULT_MAPPING_URL` or `{}` (or similar). And then, in the two functions below you could use that to collapse the `if/else` blocks into one function call with one conditional `kwarg`.

Reply:
I'll look into that.